x-pack/plugins/security/server/routes/indices/get_fields.ts

```diff
@@ -23,21 +37,41 @@ export function defineGetFieldsRoutes({ router, clusterClient }: RouteDefinition
             fields: '*',
             allowNoIndices: false,
             includeDefaults: true,
-          })) as Record<string, { mappings: Record<string, unknown> }>;
+          })) as FieldMappingResponse;
 
         // The flow is the following (see response format at https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-field-mapping.html):
         // 1. Iterate over all matched indices.
         // 2. Extract all the field names from the `mappings` field of the particular index.
-        // 3. Collect and flatten the list of the field names.
+        // 3. Collect and flatten the list of the field names, omitting any fields without mappings, and any runtime fields
         // 4. Use `Set` to get only unique field names.
+        const fields = Array.from(
```

Comment:
__question:__ how do you feel about adding a simple jest test to test this logic?

Reply:
Good idea, will do!